R/class-KPartiteNetwork.R

```diff
@@ -0,0 +1,47 @@
+check_kpartite_network <- function(object) {
+
+  errors <- character()
+
+  # Ensure that no node is in multiple partitions
+  if (length(unique(unlist(object@partitions))) < length(unlist(object@partitions))) {
+    errors <- c(errors, 'Found a node in multiple partitions. Nodes can only exist in one partition.')
+  }
+
+  # Check that all nodes are in at least one of the columns
+  if (!all(getNodeIds(object@nodes) %in% unlist(object@partitions))) {
+    errors <- c(errors, 'Found a node that is not in any partition. All nodes must be assigned to a partition.')
+  }
+
+  return(if (length(errors) == 0) TRUE else errors)
+}
+
+#' k-Partite Network 
+#' 
+#' The k-partite network class represents data in the form of a network with k distinct groups of nodes
+#' in which nodes connect only with nodes from the other groups. In other words, there are only inter-group
+#' links, no intra-group links. These k groups are commonly called partitions.
+#' k-partite networks can have any property of a regular network, but they also designate the node ids
+#' that belong to each partition (group).
+#' 
+#' @slot links LinkList object defining the links in the network.
+#' @slot nodes NodeList object defining the nodes in the network. Some nodes may not have any links.
+#' @slot linkColorScheme string defining the type of coloring scheme the links follow. Options are 'none' (default) and 'posneg'.
+#' Use a method assignLinkColors() to assign colors to links and set this slot's value.
+#' @slot partitions list of node ids that belong to each partition
+#' 
+#' @name KPartiteNetwork-class
+#' @rdname KPartiteNetwork-class
+#' @include class-network.R
+#' @export 
+KPartiteNetwork <- setClass("KPartiteNetwork", 
+  contains = "Network",
+  representation(
+    partitions = "list"
```

Comment:
This is a list of lists of strings right? Maybe we can make it a bit more formal?